### DISCUSSION TITLE
Things that require the mouse pointer

### DISCUSSION THREAD

**Discussion by LevFendi** [Factorio-Access/FactorioAccess#186] [General]
These actions seem to require that the mouse pointer is on screen and pointing at the target location:

- Using the game's built-in API-inaccessible aiming assistance feature. 

- Shooting at something. Even though the shooting state can be set with the API, it seems like the mouse pointer steals the aim back on the next tick, unless we can find a way around it. This issue is most apparent for shooting nukes.

-  ~~Using artillery remotes and spidertron remotes.~~ EDIT: These have been tested and found to work.

- ~~Using Kruise Kontrol. Until we integrate it.~~ EDIT: This has now been integrated.

Things can be added to or removed from this list as we try things out.

**Comment by EphDoering** [comment]
1 and 2 are very related. My guess is that we'd have to manage shooting_state every tick just like we do for walking state (and probably want to for repairing state). I also suspect that the auto aim will apply from the position we pass it. But if we wanted to disable the auto aim I believe we could modify raw.ammo.amo_name.ammo_type.clamped to be true in data-updates.lua. For example, the shotgun ammo doesn't auto aim and it has clamped set to true. Or conversely, if we'd like to use auto-aim I think we could set it to false. But we should do some testing to experiment, if the shooting pos in shooting_state gets set to the auto aimed entity, then the result is accessible but the input requires the mouse, and if it gets set to the mouse position, then the input is under our control and we can use the auto-aim.

For 3 What's the event / code we currently use to deploy/fire the remotes?

I know for Kruse Kontrol it's using the event position info that gets passed to Custom Input Events that request it. I think for generic mod support it would be good if we can keep the mouse over our cursor as I don't see any other way of supporting Custom Input Events that use that property. Though since you have to request the data in the custom input event prototype we could potentially warn users of incompatibility instead of it just mysteriously not working.

**Reply by LevFendi** [reply]
For (3) the relevant event name is `click-hand` or something, in `control.lua`. There is a section about capsules near the bottom of it. I might also be misremembering it...

**Reply by EphDoering** [reply]
I did the shooting state testing. It appears vanilla sets shooting position to follow the mouse exactly and auto-aim changes the actual targeted position based on that. So it appears that the actual targeted position is inaccessible but that we should be able to set the shooting position each tick to divorce this feature from the mouse.

**Comment by EphDoering** [comment]
I think we can strike capsule use from the list; we provide the position from the code.

**Reply by LevFendi** [reply]
I have tested and confirmed that they work off screen. I am striking them from the list now.
